platform/state/platform_state.proto

```diff
@@ -0,0 +1,391 @@
+     * This list MUST be deterministically ordered.
+     */
+    repeated bytes judge_hashes = 2;
+
+    /**
+     * A list of minimum judge information entries.<br/>
+     * These are "minimum ancient" entries for non-ancient rounds.
+     */
+    repeated MinimumJudgeInfo minimum_judge_info_list = 3;
+
+    /**
+     * A single consensus number.<br/>
+     * The consensus order of the next event to reach consensus.
+     */
+    uint64 next_consensus_number = 4;
+
+    /**
+     * A "consensus" timestamp.<br/>
+     * The consensus timestamp of this snapshot.
+     * <p>
+     * This SHALL be a consensus value and MAY NOT correspond to an actual
+     * "wall clock" timestamp.<br/>
+     * Consensus Timestamps SHALL always increase.
+     */
+    proto.Timestamp consensus_timestamp = 5;
+}
+
+/**
+ * Minimum ancient threshold information for round judges.
+ * This message records the minimum ancient threshold agreed by all judges in
+ * a round.
+ */
+message MinimumJudgeInfo {
+    /**
+     * A consensus round.<br/>
+     * The round this judge information applies to.
+     */
+    uint64 round = 1;
+
+    /**
+     * Minimum ancient threshold for all judges in the round.
+     * <p>
+     * This SHALL reflect the relevant minimum threshold, whether
+     * generation-based or birth-round-based.
+     */
+    uint64 minimum_judge_ancient_threshold = 2;
+}
+
+/**
+ * A network address book.<br/>
+ * The address book records the address of every known consensus node that
+ * participates in the network, including `0 weight` nodes.<br/>
+ */
+message AddressBook {
+    /**
+     * A consensus round.<br/>
+     * The round when this address book was created.
+     */
+    int64 round = 1;
+
+    /**
+     * The node ID of the next address that can be added.
+     * <p>
+     * The next node identifier assigned SHALL be equal to this value.<br/>
+     * All existing node identifiers SHALL be strictly less than this value.<br/>
+     */
+    NodeId next_node_id = 2;
+
+    /**
+     * A list of all consensus node addresses.
+     * <p>
+     * This list SHALL NOT be empty.
+     * If a consensus node is not in this list it SHALL NOT participate in the
+     * network.
+     */
+    repeated Address addresses = 3;
+}
+
+/**
+ * A single network address.
+ * This is one address in the network address book, including all required
+ * information to include that consensus node in the consensus gossip.
+ *
+ * ### Implementation Detail
+ * When `enableEventStreaming` is set to true, the memo field is REQUIRED
+ * and MUST be unique.
+ */
+message Address {
+    /**
+     * The ID of this member.
+     * <p>
+     * This identifier SHALL be agreed upon by all consensus nodes.
+     */
+    NodeId id = 1;
+
+    /**
+     * The nickname this consensus node uses to refer to another consensus node.
+     */
+    string nickname = 2;
+
+    /**
+     * The name that a consensus node uses to refer to themselves.
+     */
+    string self_name = 3;
```

Comment:
I'd recommend we move these to high field numbers. This data has no business being here, but it make take some time to fully remove all dependencies on it.

Reply:
Done.